submissions/SamVal007/Tiny-JS-OOP/index.js

```diff
@@ -0,0 +1,69 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+      Web app: https://samval007.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Person {
+  constructor(species, name, gender, speech, legs) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.speech = speech;
+    this.legs = legs;
+  }
+
+  say() {
+    const say = [
+      this.speech,
+      `I am a ${this.species}.`,
+      `My name is ${this.name}.`,
+      `My gender is ${this.gender}.`,
+      `I have ${this.legs} legs.`
+    ];
+    return say.join(' ');
+  }
+
+}
+
+class Dog extends Person {
+  constructor(name, gender, speech) {
+    super("dog", name, gender, speech, 4);
+
+  }
+  say() {
+    return `${super.say()} Pets are friendly to all!`
+  }
+}
+class Cat extends Person {
+  constructor(name, gender, speech) {
+    super("cat", name, gender, speech, 4);
+  }
+  say() {
+    return `${super.say()} Pets are friendly to all!`
+  }
```

Comment:
Code repetition. Imagine you have more kinds of pets.
Introduce the `Pet` class that will generalize shared methods from `Dog` and `Cat`.

Reply:
Ok. Bit Thanks!
If I understand correct: better to make one more class Pet which will a child of Person and and will parent for Cat and Dog. Right?:)

Reply:
Yep, the rule is whatever is shared by two classes should be generalized into their common parent class, inserted into the hyrerachy if needed.

Reply:
Ok! I got it!
Thank You!